nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncodeContent.java

```diff
@@ -55,27 +59,105 @@
             .allowableValues(BASE64_ENCODING, BASE32_ENCODING, HEX_ENCODING)
-            .defaultValue(BASE64_ENCODING)
+            .defaultValue(BASE64_ENCODING.getValue())
+            .build();
+
+    // A Boolean property descriptor that allows the user
+    // to choose if the output to encoded FlowFile content should be to a single line or multiple lines.
+    static final PropertyDescriptor SINGLE_LINE_OUTPUT = new PropertyDescriptor.Builder()
+            .name("single-line-output")
+            .displayName("Output Content to Single Line")
+            .description("If set to 'true', the encoded FlowFile content will output as a single line. If set to 'false', "
+                + "it will output as multiple lines. This property is only applicable when Base64 or Base32 encoding is selected.")
+            .required(false)
+            .defaultValue("false")
+            .allowableValues(SINGLE_LINE_OUTPUT_TRUE, SINGLE_LINE_OUTPUT_FALSE)
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .dependsOn(MODE, ENCODE_MODE)
+            .dependsOn(ENCODING, BASE64_ENCODING, BASE32_ENCODING)
+            .build();
+
+    static final PropertyDescriptor ENCODED_LINE_SEPARATOR = new PropertyDescriptor.Builder()
+        .name("line-separator")
+        .displayName("Encoded Content Line Separator")
+        .description("Each line of encoded data will be terminated with this byte sequence (e.g. \\r\\n"
+                + "). This property defaults to the system-dependent line separator string.  If `line-length` <= 0, "
+                + "the `line-separator` property is not used. This property is not used for `hex` encoding.")
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .defaultValue(System.lineSeparator())
+        .addValidator(Validator.VALID)
+        .dependsOn(MODE, ENCODE_MODE)
+        .dependsOn(ENCODING, BASE64_ENCODING, BASE32_ENCODING)
+        .build();
+
+    static final PropertyDescriptor ENCODED_LINE_LENGTH = new PropertyDescriptor.Builder()
+        .name("encoded-line-length")
+        .displayName("Encoded Content Line Length")
+        .description("Each line of encoded data will contain `encoded-line-length` characters (rounded down to the nearest multiple of 4). "
+            + "If `encoded-line-length` <= 0, the encoded data is not divided into lines. This property is "
+            + "ignored if `single-line-output` is set to True.")
+        .required(false)
+        .defaultValue("76")
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .addValidator(StandardValidators.INTEGER_VALIDATOR)
+        .dependsOn(MODE, ENCODE_MODE)
+        .dependsOn(ENCODING, BASE64_ENCODING, BASE32_ENCODING)
+        .dependsOn(SINGLE_LINE_OUTPUT, SINGLE_LINE_OUTPUT_FALSE)
+        .build();
+
+    static final PropertyDescriptor FAIL_ON_ZERO_LENGTH_CONTENT = new PropertyDescriptor.Builder()
```

Comment:
Is there a reason for adding this particular property? It seems better to avoid introducing it, as other components can handle routing on file size in a standard flow configuration.

Reply:
Done.  This is removed.